plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/SelfLoopPreProcessor.java

```diff
@@ -74,7 +74,9 @@ public void process(final LGraph layeredGraph, final IElkProgressMonitor monitor
                 preprocessLabels(slNode, layoutDirection);
                 
                 // hide the ports which are non useful for the crossing minimization
-                hidePorts(node);
+                if (!node.getProperty(LayeredOptions.PORT_CONSTRAINTS).isPosFixed()) {
```

Comment:
I'm not entirely sure whether this violates assumptions later on, but it does seem appropriate and works with all of my test models, so I vote for merging the pull request.